addons/debugger/userscript.js

```diff
@@ -0,0 +1,296 @@
+    }
+    if (consoleWrapper.offsetLeft - pos1 <= maxX && consoleWrapper.offsetLeft - pos1 >= 0) {
+      consoleWrapper.style.left = consoleWrapper.offsetLeft - pos1 + "px";
+    }
+  }
+
+  function closeDragElement() {
+    // stop moving when mouse button is released:
+    document.removeEventListener("mouseup", closeDragElement);
+    document.removeEventListener("mousemove", elementDrag);
+  }
+
+  trashButton.addEventListener("click", () => {
+    document.querySelectorAll(".log").forEach((log, i) => log.remove());
+    closeDragElement();
+    logs = [];
+  });
+  trashButton.addEventListener("mouseup", () => {
+    closeDragElement();
+  });
+  closeButton.addEventListener("click", () => toggleConsole(false));
+  closeButton.addEventListener("mouseup", () => closeDragElement());
+  let download = (filename, text) => {
+    var element = document.createElement("a");
+    element.setAttribute("href", "data:text/plain;charset=utf-8," + encodeURIComponent(text));
+    element.setAttribute("download", filename);
+
+    element.style.display = "none";
+    document.body.appendChild(element);
+
+    element.click();
+
+    document.body.removeChild(element);
+  };
+
+  exportButton.addEventListener("click", () => {
+    closeDragElement();
+    let file = logs.join("\n");
+    download("logs.txt", file);
+  });
+  let logs = [];
+  const addItem = (content, thread, type) => {
+    workspace = Blockly.getMainWorkspace();
+    const wrapper = document.createElement("div");
+    const span = (text, cl = "") => {
+      let s = document.createElement("span");
+      s.innerText = text;
+      s.className = cl;
+      return s;
+    };
+
+    const scrolledDown = extraContainer.scrollTop + 5 > extraContainer.scrollHeight - extraContainer.clientHeight;
+
+    const targetName = thread.target.getName();
+    wrapper.className = "log";
+    wrapper.classList.add(type);
+    consoleList.append(wrapper);
+
+    const blockId = thread.peekStack();
+    const block = workspace.getBlockById(blockId);
+    const inputBlock = block.getChildren().find((b) => b.parentBlock_.id === blockId);
+    if (inputBlock.type != "text") {
+      if (inputBlock.inputList.filter((i) => i.name).length === 0) {
+        const inputSpan = document.createElement("span");
+        const inputBlockFill = getComputedStyle(inputBlock.svgPath_).fill;
+        const inputBlockStroke = getComputedStyle(inputBlock.svgPath_).stroke;
+        // for compatibility with custom block colors
+        const inputBlockColor =
+          inputBlockFill == "rgb(40, 40, 40)" || inputBlockFill == "rgb(255, 255, 255)"
+            ? inputBlockStroke
+            : inputBlockFill;
+        inputSpan.innerText = inputBlock.toString();
+        inputSpan.className = "console-variable";
+        inputSpan.style.background = inputBlockColor;
+        wrapper.append(inputSpan);
+      }
+    }
+    let string = addon.settings
+      .get("exportFormat")
+      .replace("{sprite}", targetName)
```

Comment:
This may log incorrect stuff for sprites named {content}. Consider regex.

Reply:
Wouldn't replacing content first fix this?